pkg/scalers/nats_jetstream_scaler.go

```diff
@@ -149,15 +146,34 @@ func parseNATSJetStreamMetadata(config *ScalerConfig) (natsJetStreamMetadata, er
 	}
 
 	meta.scalerIndex = config.ScalerIndex
+
+	natsServerEndpoint, err := GetFromAuthOrMeta(config, "natsServerMonitoringEndpoint")
+	if err != nil {
+		return meta, err
+	}
+	useHTTPS := false
+	if val, ok := config.TriggerMetadata["useHttps"]; ok {
+		useHTTPS, err = strconv.ParseBool(val)
+		if err != nil {
+			return meta, fmt.Errorf("useHTTPS parsing error %s", err.Error())
+		}
+	}
+	meta.monitoringEndpoint = getNATSJetStreamEndpoint(useHTTPS, natsServerEndpoint, meta.account)
+
 	return meta, nil
 }
 
-func (s *natsJetStreamScaler) getNATSJetStreamEndpoint() string {
-	return fmt.Sprintf("http://%s/jsz?acc=%s&consumers=true&config=true", s.metadata.monitoringEndpoint, s.metadata.account)
+func getNATSJetStreamEndpoint(useHTTPS bool, natsServerEndpoint string, account string) string {
+	protocol := natsHTTPProtocol
+	if useHTTPS {
+		protocol = natsHTTPSProtocol
+	}
+
+	return fmt.Sprintf("%s://%s/jsz?acc=%s&consumers=true&config=true", protocol, natsServerEndpoint, account)
 }
 
 func (s *natsJetStreamScaler) IsActive(ctx context.Context) (bool, error) {
-	monitoringEndpoint := s.getNATSJetStreamEndpoint()
+	monitoringEndpoint := s.metadata.monitoringEndpoint
```

Comment:
Why don't use `s.metadata.monitoringEndpoint` directly? I mean, there is only 1 place where `monitoringEndpoint` is used